services/wcr/legal_entities.md

```diff
@@ -0,0 +1,62 @@
+# Legal entity names and Business names
+
+The users of the service - carriers, brokers and dealers, hereafter referred to as "entities" - can have either or both a legal entity name or an informal business name. At least one of these must be present in all cases. The combinations allowed depend on the business type of the entity and the tier of the registration. In some cases, fields in the database may be populated or nil depending on whether the registration was created before or after the deployment of new legal entities logic (ca. April 2022).
+
+## Business logic
+
+### Legal entity names
+The legal entity name of an entity is the formally registered name of the entity, if any. The business logic for legal entity names varies according to the business type, tier and the date of the registration.
+- Limited Companies and Limited Liability Partnerships:
+  - Upper-tier: The legal entity name is the registered company name retrieved from Companies House during the registration, e.g. "Applied Waste Solutions Limited". Note that this will not be populated for registrations created before the deployment of the legal entities changes.
+  - Lower-tier: Currently nil, as companies house deteails are not retrieved for lower-tier registrations.
+- Sole Traders:
+  - Upper-tier: The legal entity name is the first name and last name of the key person for the registration, e.g. "Jim Smith". This is always populated.
+  - Lower-tier: Currently nil, as the key person name is not stored for lower-tier sole traders.
+- Other:
+  - Legal entity name is nil in all cases.
+
+### Business names
```

Comment:
Can this match the attribute so that when we look at the code we know exactly what this refers to? ie `company_name`